src/trade/spread.ts

```diff
@@ -25,16 +106,39 @@ export const getSpreadWithPriceImpactP = (
       getCurrentOiWindowId(oiWindowsSettings),
       oiWindowsSettings.windowsCount,
       oiWindows,
-      buy
+      spreadCtx?.isOpen !== false ? buy : !buy
     );
   }
 
   if (!onePercentDepth || activeOi === undefined || collateral === undefined) {
-    return pairSpreadP;
+    return pairSpreadP / 2;
   }
 
   return (
-    pairSpreadP +
-    (activeOi + (collateral * leverage) / 2) / onePercentDepth / 100
+    getSpreadP(pairSpreadP) +
+    ((activeOi * getCumulativeFactor(spreadCtx) + (collateral * leverage) / 2) /
+      onePercentDepth /
+      100 /
+      getLegacyFactor(spreadCtx)) *
+      getProtectionCloseFactor(spreadCtx)
   );
 };
+
+export const getSpreadP = (
```

Comment:
Surprised we don't pass `open` here and `contractsVersion` because when adding to a trade opened before v9.2 the opening spread is still 100% and when closing a trade opened before v9.2 the closing spread is 0.